akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/EventSourcedBehavior.scala

```diff
@@ -156,6 +156,8 @@ object EventSourcedBehavior {
    *
    * You may configure the behavior to skip replaying snapshots completely, in which case the recovery will be
    * performed by replaying all events -- which may take a long time.
+   *
+   * Overrides previously set [[EventSourcedBehavior.withRecovery]]
    */
   def withSnapshotSelectionCriteria(selection: SnapshotSelectionCriteria): EventSourcedBehavior[Command, Event, State]
```

Comment:
It's tempting to move this into a new `Recovery` strategy.

```
withRecovery(Recovery.withSnapshotSelectionCriteria(selection))
```

Would require deprecation of this.

wdyt?

Reply:
Yep great idea. That would leave no space for overriding combinations of those two